src/backend/gl/src/device.rs

```diff
@@ -116,6 +119,7 @@ impl Device {
     pub(crate) fn new(share: Starc<Share>) -> Self {
         Device {
             share: share,
+            id: d::DeviceId(thread_rng().gen::<usize>()),
```

Comment:
Maybe something like [`snowflake`](https://github.com/Stebalien/snowflake) would be more appropriate?